### PR TITLE
[SHELL32] Browse for Folder: added keyboard navigation

### PR DESCRIPTION
**Pull request by Gathub22**
Added patches to buttons in folder browsing menu's buttons to fix [CORE-19570](https://jira.reactos.org/browse/CORE-19570).

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
Approved. Maybe some WS_TABSTOP could be stripped additionally, but that future possible improvement doesn't make any of your current changes wrong. lgtm.

**Review by binarymaster — DISMISSED** [review]
Hello,
Thanks for your interest in ReactOS!
We ask our contributors [to use their full name when committing](https://github.com/reactos/reactos/blob/master/CONTRIBUTING.md#rules-and-recommendations).
[Please amend your commit with your full name](https://stackoverflow.com/a/3042512), and update this PR.

To change this for future PR's, you can update it for globally (for the system):
```
git config --global user.name "Your Name"
git config --global user.email you@example.com
```

Or just for the current repository:
```
git config user.name "Your Name"
git config user.email you@example.com
```

Also please set up your GitHub account accordingly, so maintainers can use Squash & Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:

1. Set your full name in [GitHub profile settings](https://github.com/settings/profile) (it should match your `user.name` setting)
1. Uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails)

Current commit state:
image

Regards,

**Review by HBelusca — APPROVED** [review]

**Comment by Gathub22** [comment]
> Hello, Thanks for your interest in ReactOS! We ask our contributors [to use their full name when committing](https://github.com/reactos/reactos/blob/master/CONTRIBUTING.md#rules-and-recommendations). [Please amend your commit with your full name](https://stackoverflow.com/a/3042512), and update this PR.
> 
> To change this for future PR's, you can update it for globally (for the system):
> 
> ```
> git config --global user.name "Your Name"
> git config --global user.email you@example.com
> ```
> 
> Or just for the current repository:
> 
> ```
> git config user.name "Your Name"
> git config user.email you@example.com
> ```
> 
> Also please set up your GitHub account accordingly, so maintainers can use Squash & Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:
> 
>     1. Set your full name in [GitHub profile settings](https://github.com/settings/profile) (it should match your `user.name` setting)
> 
>     2. Uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails)
> 
> 
> Current commit state: image
> 
> Regards,

Thank you for telling me! I haven't noticed the md file with the rules. My bad, sorry. I've fixed my profile name, my Git name and the commit is now fixed and pushed. Tell me if it's okay :)

**Comment by binarymaster** [comment]
> Thank you for telling me! I haven't noticed the md file with the rules. My bad, sorry. I've fixed my profile name, my Git name and the commit is now fixed and pushed. Tell me if it's okay :)

Looks fine now, thanks!

File: dll/win32/shell32/lang/en-US.rc
Comment (HBelusca) [inline dll/win32/shell32/lang/en-US.rc:113]: Note that the `DEFPUSHBUTON` "type" (the 1st thing on the left), already includes `BS_DEFPUSHBUTTON` and `WS_TABSTOP` styles, see https://learn.microsoft.com/en-us/windows/win32/menurc/defpushbutton-control
You can therefore delete the whole `BS_DEFPUSHBUTTON | WS_TABSTOP` on the right.

File: dll/win32/shell32/lang/en-US.rc
Comment (HBelusca) [inline dll/win32/shell32/lang/en-US.rc:114]: Similarly, the `PUSHBUTTON` already contains `WS_TABSTOP`, so that one too can be removed
https://learn.microsoft.com/en-us/windows/win32/menurc/pushbutton-control

**Review by HBelusca — CHANGES_REQUESTED** [review]

**Comment by HBelusca** [comment]
Cc @maj113 too: In your PR #6833 (commit 7ce96fd36) your `PUSHBUTTON`s have the residual explicit `WS_TABSTOP` style that is unnecessary.

**Comment by maj113** [comment]
Thanks for letting me know, will fix.

File: dll/win32/shell32/lang/en-US.rc
Comment (Gathub22) [reply dll/win32/shell32/lang/en-US.rc:114]: > Similarly, the `PUSHBUTTON` already contains `WS_TABSTOP`, so that one too can be removed https://learn.microsoft.com/en-us/windows/win32/menurc/pushbutton-control

So basically those styles are unnecesary and the buttons should look like this?
```
DEFPUSHBUTTON "OK", IDOK, 102, 174, 50, 14
PUSHBUTTON "Cancel", IDCANCEL, 156, 174, 50, 14
```

Comment (JoachimHenze) [reply dll/win32/shell32/lang/en-US.rc:114]: The thing is that we most likely do not want IDOK to have a WS_TABSTOP (which it would have by default as a DEFPUSHBUTTON).
Since there is a   PUSHBUTTON "&Make New Folder" at the line before it, and we want to be able to navigate between these 3 buttons with the arrow keys. They are in the same group.

I would therefore recommed to use
```
DEFPUSHBUTTON "OK", IDOK, 102, 174, 50, 14, BS_DEFPUSHBUTTON
PUSHBUTTON "Cancel", IDCANCEL, 156, 174, 50, 14
```
instead. Or (if that wouldn't work yet) we could use:
```
DEFPUSHBUTTON "OK", IDOK, 102, 174, 50, 14, BS_DEFPUSHBUTTON | NOT WS_TABSTOP
PUSHBUTTON "Cancel", IDCANCEL, 156, 174, 50, 14
```

That should behave better than:
```
PUSHBUTTON "&Make New Folder", IDC_BROWSE_FOR_FOLDER_NEW_FOLDER, 12, 174, 77, 14, WS_GROUP | WS_TABSTOP
DEFPUSHBUTTON "OK", IDOK, 102, 174, 50, 14
PUSHBUTTON "Cancel", IDCANCEL, 156, 174, 50, 14
```

But like I said: That was already broken before your changes.

Comment (JoachimHenze) [reply dll/win32/shell32/lang/en-US.rc:114]: Don't strip any WS_TABSTOP here. **The PR is correct like it currently is.**
I tested on WinXP the browseForFolderDialog by starting rapps and navigating into its properties.
**All 3 buttons seem to have a WS_TABSTOP.**
I can tabtabtab and it will select each of it then.
But still I can use the arrow-keys to navigate between them which means only the first one has a WS_GROUP style.

**The PR can be merged as is and should not be changed.**

Comment (Gathub22) [reply dll/win32/shell32/lang/en-US.rc]: No problem. Those styles have been cleaned and the menu works fine :)

Comment (HBelusca) [reply dll/win32/shell32/lang/en-US.rc]: @JoachimHenze
>  I tested on WinXP the browseForFolderDialog by starting rapps and navigating into its properties.
All 3 buttons seem to have a WS_TABSTOP.

`PUSHBUTTON` _**already**_ contains `WS_TABSTOP`. You see it in the resource because you've also `CONTROL xxxx "Button", ..., BS_PUSHBUTTON | WS_TABSTOP`, which is the resource expansion of `PUSHBUTTON`.

File: dll/win32/shell32/lang/bg-BG.rc
Comment (HBelusca) [inline dll/win32/shell32/lang/bg-BG.rc:96]:
```suggestion
    DEFPUSHBUTTON "Добре", 1, 80, 176, 50, 12, WS_GROUP
```
These two removed styles are already in `DEFPUSHBUTTON`

File: dll/win32/shell32/lang/bg-BG.rc
Comment (HBelusca) [inline dll/win32/shell32/lang/bg-BG.rc:97]:
```suggestion
    PUSHBUTTON "Отказ", 2, 134, 176, 50, 12
```

**Review by HBelusca — CHANGES_REQUESTED** [review]
and same changes in the other languages.

File: dll/win32/shell32/lang/bg-BG.rc
Comment (JoachimHenze) [reply dll/win32/shell32/lang/bg-BG.rc:96]: No don't do that! That suggestion is wrong @HBelusca . You would lose the tabstop then (due to the extended syle WS_GROUP being set here explicitly!). And that is wrong!

File: dll/win32/shell32/lang/bg-BG.rc
Comment (JoachimHenze) [reply dll/win32/shell32/lang/bg-BG.rc:97]: For the record: That is an unfair request @HBelusca ! He didn't touch that line. `And the whole file is full of that crap!` Let's better clean up the whole files instead **with a future PR**!

File: dll/win32/shell32/lang/en-US.rc
Comment (JoachimHenze) [reply dll/win32/shell32/lang/en-US.rc]: Yes, I do understand that.

File: dll/win32/shell32/lang/bg-BG.rc
Comment (HBelusca) [reply dll/win32/shell32/lang/bg-BG.rc:97]: OK then @Gathub22 can fix it only for the lines he originally had to modify to implement the intended fix for the PR.

File: dll/win32/shell32/lang/bg-BG.rc
Comment (Gathub22) [reply dll/win32/shell32/lang/bg-BG.rc:96]: Im okay with cleaning my code. However, I can't find those buttons to test how the navigation goes. Where are these?

Comment (Gathub22) [reply dll/win32/shell32/lang/bg-BG.rc:96]: However @JoachimHenze is right too. For just a simple fix, we could clean and properly test all styles later in another PR and avoid delaying this one

File: dll/win32/shell32/lang/bg-BG.rc
Comment (JoachimHenze) [reply dll/win32/shell32/lang/bg-BG.rc:97]: Ok, I will merge yours now. Please wait with opening another shell32 rc PR for 2 weeks, we do have at least 2 more PRs which are currently waiting for a merge, and which would collide with such a widespread diff in the RCs.
Act after merging them, one is a ro-RO specific one, the other touches all rcs. I will merge both, in the next 2 weeks most likely.

**Pull request merged**
